### PR TITLE
[SPARK-7997][Core]Remove Akka from Spark Core and Streaming

### PR DESCRIPTION
**Pull request by zsxwing**
- Remove Akka dependency from core. Note: the streaming-akka project still uses Akka.
- Remove HttpFileServer
- Remove Akka configs from SparkConf and SSLOptions
- Rename `spark.akka.frameSize` to `spark.rpc.message.maxSize`. I think it's still worth to keep this config because using `DirectTaskResult` or `IndirectTaskResult`  depends on it.
- Update comments and docs

### CODE REVIEW

**Comment by zsxwing** [comment]
CC @rxin @vanzin

**Comment by zsxwing** [comment]
retest this please

**Comment by zsxwing** [comment]
Looks weird. Not sure why the following errors didn't happen in #10744

```
[error]  * the type hierarchy of class org.apache.spark.streaming.zeromq.ZeroMQReceiver has changed in new version. Missing types {org.apache.spark.streaming.receiver.ActorHelper}
[error]    filter with: ProblemFilters.exclude[MissingTypesProblem]("org.apache.spark.streaming.zeromq.ZeroMQReceiver")
[info] spark-streaming: found 1 potential binary incompatibilities (filtered 37)
[error]  * class org.apache.spark.streaming.receiver.ActorReceiver#Supervisor does not have a correspondent in new version
[error]    filter with: ProblemFilters.exclude[MissingClassProblem]("org.apache.spark.streaming.receiver.ActorReceiver$Supervisor")
```

**Comment by rxin** [comment]
LGTM

**Comment by zsxwing** [comment]
retest this please

File: core/src/main/scala/org/apache/spark/rpc/netty/Dispatcher.scala [hunk 1]
```diff
@@ -175,7 +175,7 @@ private[netty] class Dispatcher(nettyEnv: NettyRpcEnv) extends Logging {
     endpoints.keySet().asScala.foreach(unregisterRpcEndpoint)
     // Enqueue a message that tells the message loops to stop.
     receivers.offer(PoisonPill)
-    threadpool.shutdown()
+    threadpool.shutdownNow()
```
Comment (zsxwing) [inline core/src/main/scala/org/apache/spark/rpc/netty/Dispatcher.scala]: Just found here didn't interrupt threads previously.

**Comment by zsxwing** [comment]
> Test build #49860 has finished for PR 10854 at commit 39f21de.
> 
> This patch fails from timeout after a configured wait of `250m`.
> This patch merges cleanly.
> This patch adds no public classes.

Looks a netty bug. I just set the netty3 version to 3.8.0.Final and then passed the flume tests locally. Since we have been used 3.8.0.Final for a long time, let's keep it rather than downgrading it.

**Comment by zsxwing** [comment]
retest this please

**Comment by rxin** [comment]
I'm going to merge this. Not sure why Jenkins reported timeout. I looked at the Jenkins logs and tests all passed.

**Pull request closed without merging**